dspy/adapters/json_adapter.py

```diff
@@ -303,3 +316,50 @@ def format_signature_fields_for_instructions(role, fields: Dict[str, FieldInfo])
     #              ", and then ending with the marker for `completed`.")
 
     return "\n\n".join(parts).strip()
+
+
+def _get_structured_outputs_response_format(signature: SignatureMeta) -> pydantic.BaseModel:
+    """
+    Obtains the LiteLLM / OpenAI `response_format` parameter for generating structured outputs from
+    an LM request, based on the output fields of the specified DSPy signature.
+
+    Args:
+        signature: The DSPy signature for which to obtain the `response_format` request parameter.
+    Returns:
+        A Pydantic model representing the `response_format` parameter for the LM request.
+    """
+
+    def filter_json_schema_extra(field_name: str, field_info: FieldInfo) -> FieldInfo:
+        """
+        Recursively filter the `json_schema_extra` of a FieldInfo to exclude DSPy internal attributes
+        (e.g. `__dspy_field_type`) and remove descriptions that are placeholders for the field name.
+        """
+        field_copy = deepcopy(field_info)  # Make a copy to avoid mutating the original
+
+        # Update `json_schema_extra` for the copied field
+        if field_copy.json_schema_extra:
+            field_copy.json_schema_extra = {
+                key: value
+                for key, value in field_info.json_schema_extra.items()
+                if key not in ("desc", "__dspy_field_type")
+            }
+            field_desc = field_info.json_schema_extra.get("desc")
+            if field_desc is not None and field_desc != f"${{{field_name}}}":
+                field_copy.json_schema_extra["desc"] = field_desc
+
+        # Handle nested fields
+        if hasattr(field_copy.annotation, "__pydantic_model__"):
+            # Recursively update fields of the nested model
+            nested_model = field_copy.annotation.__pydantic_model__
+            updated_fields = {
+                key: filter_json_schema_extra(key, value) for key, value in nested_model.__fields__.items()
```

Comment:
Curious - why do we need recursive handling? `nested_model.__fields__` should be Pydantic fields instead of DSPy fields, do they also have these DSPy internal attributes like `__dspy_field_type`?

Reply:
In the vast majority of cases, hopefully not....

Though the following is valid:

```
import dspy

import pydantic

class Obj(pydantic.BaseModel):
    a: int = dspy.OutputField()
    b: str

class MySig(dspy.Signature):
    inp: str = dspy.InputField() 
    outp: Obj = dspy.OutputField()

print(MySig.schema())
```